src/unix/fs.c

```diff
@@ -1053,10 +1053,83 @@ static void uv__to_stat(struct stat* src, uv_stat_t* dst) {
 }
 
 
+static int uv__fs_statx(int fd,
+                        const char* path,
+                        int is_fstat,
+                        int is_lstat,
+                        uv_stat_t* buf) {
+  STATIC_ASSERT(UV_ENOSYS != -1);
+#ifdef __linux__
+  static int no_statx;
+  struct uv__statx statxbuf;
+  int dirfd;
+  int flags;
+  int mode;
+  int rc;
+
+  if (no_statx)
+    return UV_ENOSYS;
+
+  dirfd = -100; /* AT_FDCWD */
+  flags = 0; /* AT_STATX_SYNC_AS_STAT */
+  mode = 0xFFF; /* STATX_BASIC_STATS + STATX_BTIME */
+
+  if (is_fstat) {
+    dirfd = fd;
+    flags |= 0x1000; /* AT_EMPTY_PATH */
+  }
+
+  if (is_lstat)
+    flags |= 0x100; /* AT_SYMLINK_NOFOLLOW */
+
+  rc = uv__statx(dirfd, path, flags, mode, &statxbuf);
+
+  if (rc == -1) {
+    /* EPERM only seems to happen when running inside a Docker container
+     * and is almost certainly indicative of a Docker bug because there
+     * are no kernel code paths that can reasonably result in EPERM.
+     */
```

Comment:
I'm 80% positive it's because of docker's seccomp filter but I'm not invested enough to investigate in more detail. :-)

Reply:
Because I don't like leaving loose ends: upgrading to docker 18.04 and libbseccomp 2.3.3 fixes the EPERM error. I'll update the comment before merging the PR.

The CI's docker guests run on an ubuntu 16.04 host that has libseccomp 2.3.1 installed and doesn't know about the statx system call.